packages/ckeditor5-bookmark/src/bookmarkui.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module bookmark/bookmarkui
+ */
+
+import { Plugin } from 'ckeditor5/src/core.js';
+import { ButtonView, MenuBarMenuListItemButtonView } from 'ckeditor5/src/ui.js';
+
+import bookmarkIcon from '../theme/icons/bookmark.svg';
+import '../theme/bookmark.css';
+
+/**
+ * The UI plugin of the bookmark feature.
+ *
+ * It registers the `'bookmark'` UI button in the editor's {@link module:ui/componentfactory~ComponentFactory component factory}
+ * which inserts the `bookmark` element upon selection.
+ */
+export default class BookmarkUI extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	public static get pluginName() {
+		return 'BookmarkUI' as const;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public init(): void {
+		const editor = this.editor;
+
+		editor.ui.componentFactory.add( 'bookmark', () => {
+			const buttonView = this._createButton( ButtonView );
+
+			buttonView.set( {
+				tooltip: true
```

Comment:
Lets add the `bookmarkIcon` to the button.

Reply:
In that scenario the icon will not be visible in menu bar, so it will require to add icon in another place as well

Reply:
Correct. Bookmark icon should be also visible in menu bar.